inference_engine/src/controller.py

```diff
@@ -95,22 +95,17 @@ def _check_pdf(exam_id, file_name, pdf_type: PDFType, force=False):
             return
         for i, image in enumerate(images):
             try:
-                results = _check_image(image)
-            except ExamNotDetected as e:
+                results = _check_image(image, pdf_type == PDFType.answer_sheets)
+            except ExamInvalid as e:
                 logging.warning(f"Error during processing page {i + 1} in file {file_name}: {e}")
-                output_dir = tmp_dir / "unknown"
+                output_dir = tmp_dir / e.FILENAME
                 output_dir.mkdir(exist_ok=True)
                 image.save(f"{output_dir}/unknown_{i}.jpg", "JPEG")
                 continue
             output_dir = tmp_dir
             sufix = ""
             if pdf_type == PDFType.answer_sheets:
                 output_dir = output_dir / results.student_id_boxes
-                version = versioning.get_next_version(
-                    output_dir,
-                    remote_storage.get_student_dir(exam_id, index=results.student_id_boxes),
-                )
-            elif pdf_type == PDFType.answer_keys:
```

Comment:
@marekhering this breaks handling of both students answers and answers keys. This hunk should be restored